### PR TITLE
feature: dfdaemon supports hot reload

### PR DESCRIPTION
**Pull request by SataQiu**
Signed-off-by: SataQiu <1527062125@qq.com>

### Ⅰ. Describe what this PR did
feature: dfdaemon supports hot reload

### CODE REVIEW

**Comment by Starnop** [comment]
Could you please add more descriptions about why to submit this PR?

File: cmd/dfdaemon/main.go [hunk 1]
```diff
@@ -17,9 +17,27 @@
...
 	"github.com/dragonflyoss/Dragonfly/cmd/dfdaemon/app"
+	dferr "github.com/dragonflyoss/Dragonfly/pkg/errortypes"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
 func main() {
-	app.Execute()
+	rand.Seed(time.Now().UnixNano())
```
Comment (Starnop) [inline cmd/dfdaemon/main.go]: What is the meaning of this line of code?

File: cmd/dfdaemon/main.go [hunk 2]
```diff
@@ -17,9 +17,27 @@
...
+
+	command := app.NewDFDaemonCommand()
+
```
Comment (Starnop) [inline cmd/dfdaemon/main.go]: delete this blank line?

File: pkg/cmd/version.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/dragonflyoss/Dragonfly/version"
+
+	"github.com/spf13/cobra"
+)
+
+// versionLongTempl is used to describe version command in detail and auto generate command doc.
+var versionLongTempl = "Display the version and build information of Dragonfly %s, " +
+	"including GoVersion, OS, Arch, Version, BuildDate and GitCommit."
+
+func NewVersionCommand(name string) *cobra.Command {
```
Comment (allencloud) [inline pkg/cmd/version.go]: Please do not forget to add the `Example` field. And it is be used to generate docs.

File: cmd/dfdaemon/app/version.go [hunk 4]
```diff
@@ -1,56 +0,0 @@
...
-	Use:           "version",
-	Short:         "Show the current version of dfdaemon",
-	Long:          versionDescription,
-	SilenceErrors: true,
-	SilenceUsage:  true,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		printer.Printf(version.Print("dfdaemon"))
-		return nil
-	},
-	Example: versionExample(),
```
Comment (allencloud) [inline cmd/dfdaemon/app/version.go]: see here.

**Comment by inoc603** [comment]
I think the `options` package may not be very necessary. It's only used by the `dfdaemon` command.  It has a single file with only 2 functions, and the filename `generic.go` suggests that the purpose of that file is kind of unclear. Setting flags and parsing command line arguments are tightly coupled with the command itself, so it might be better to keep them where they're used.

File: cmd/dfdaemon/main.go (hunk 1 above)
Comment (zhouhaibing089) [reply cmd/dfdaemon/main.go]: See https://golang.org/pkg/math/rand/#Rand.Seed.

**Comment by SataQiu** [comment]
Hi @Starnop What do you think about this PR?

**Comment by SataQiu** [comment]
> very nice work! one suggestion: `dfdaemon` should check the changed config file before stopping, it cannot stop if the new config is invalid.

Done!

**Comment by SataQiu** [comment]
I'm going to close this PR now.
Maybe we can do what kube-proxy has done: exit the program after detecting a configuration change and let k8s pull up the program again automatically to apply the new configuration.
Ref: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server.go#L250-L275

/close

**Pull request closed without merging**